code/game/objects/items/weapons/holy_weapons.dm

```diff
@@ -301,6 +327,67 @@
 		qdel(S)
 	return ..()
 
+/obj/item/nullrod/scythe/talking/attackby(obj/item/I, mob/user, params)
+	if(!istype(I, /obj/item/soulstone) || !possessed)
+		return ..()
+	if(obj_integrity >= max_integrity)
+		to_chat(user, "<span class='notice'>You have no reason to replace a perfectly good soulstone with a new one.</span>")
+		return
+	to_chat(user, "<span class='notice'>You load a new soulstone into the possessed blade.</span>")
+	playsound(user, 'sound/weapons/gun_interactions/shotgunpump.ogg', 60, TRUE)
+	obj_integrity = max_integrity
+	for(var/mob/living/simple_animal/shade/sword/sword_shade in contents)
+		sword_shade.health = sword_shade.maxHealth
+	qdel(I)
+
+/obj/item/nullrod/scythe/talking/take_damage(damage_amount)
+	if(possessed)
+		for(var/mob/living/simple_animal/shade/sword/sword_shade in contents)
+			sword_shade.take_overall_damage(damage_amount)
+	return ..()
+
+/obj/item/nullrod/scythe/talking/proc/click_actions(atom/attacking_atom, mob/living/simple_animal/attacking_shade)
+	if(world.time <= attacking_shade.next_move) // yea we gotta check
+		return
+	if(!ismovable(attacking_atom))
+		return
+	attacking_shade.changeNext_move(CLICK_CD_MELEE)
+	if(ismob(loc))
```

Comment:
```suggestion
	if(ishuman(loc))
```
should typecheck to what var typepath you're using

Reply:
changed